src/components/time-slider.tsx

```diff
@@ -25,11 +42,25 @@ function TimeSlider() {
     setValue(Number(e.target.value))
   }, [setValue])
 
+  const togglePlayPause = () => {
+    if (isPlaying) {
+      setIsPlaying(false)
+    } else {
+      if (value === lastDateIndex) {
+        setValue(0)
+      }
+      setIsPlaying(true)
+    }
+  }
+
   return (
     <div className="time-slider">
+      <button onClick={togglePlayPause}>{isPlaying ? "Pause" : "Play"}</button>
```

Comment:
After playing with this I think we need to set a width for the button, otherwise the centered content shifts when the label switches from "Pause" to "Play".  We may also want to move button to the right of the slider but we can leave that to Kate's preference.